src/atama/head/head.cpp

```diff
@@ -0,0 +1,576 @@
+            snprintf(buffer, sizeof(buffer), "pan_%d_tilt_%d", 0, i);
+            val.at(buffer).get_to(tilt_to_distance_[i]);
+          }
+        } catch (nlohmann::json::parse_error & ex) {
+          std::cerr << "parse error at byte " << ex.byte << std::endl;
+        }
+      } else if (key == "PanDistanceToTilt") {
+        try {
+          for (int i = 4; i >= 0; i--) {
+            for (int j = 0; j <= i; j++) {
+              char buffer[32];
+              snprintf(buffer, sizeof(buffer), "pan_%d_distance_%d", (i - j), j);
+              val.at(buffer).get_to(pan_distance_to_tilt_[i - j][j]);
+            }
+          }
+        } catch (nlohmann::json::parse_error & ex) {
+          std::cerr << "parse error at byte " << ex.byte << std::endl;
+        }
+      }
+    }
+  } catch (nlohmann::json::parse_error & ex) {
+    // TODO(nathan): will be used for logging
+  }
+}
+
+void Head::track_object(std::string object_name)
+{
+  if (!check_time_belief()) {
+    return;
+  }
+
+  function_id = Head::TRACK_OBJECT;
+  stop_scan();
+
+  // filter the object by its label
+  std::vector<ninshiki_interfaces::msg::DetectedObject> filtered_result;
+  if (!detection_result.empty()) {
+    for (const auto & item : detection_result) {
+      if (item.label == object_name) {
+        filtered_result.push_back(item);
+      }
+    }
+  } else {
+    return;
+  }
+
+  // looking at the center of the object
+  // We pick an object with the biggest confidence
+  double confidence = 0.0;
+  double object_center_x, object_center_y;
+  int camera_width, camera_height;
+  if (!filtered_result.empty()) {
+    for (const auto & item : detection_result) {
+      if (item.score > confidence) {
+        confidence = item.score;
+        camera_width = item.img_width;
+        camera_height = item.img_height;
+
+        object_center_x = (item.left * camera_width +
+          item.right * camera_width) / 2;
+        object_center_y = (item.top * camera_height +
+          item.bottom * camera_height) / 2;
+      }
+    }
+  } else {
+    return;
+  }
+  keisan::Point2 pos = keisan::Point2(object_center_x, object_center_y);
+
+  float diagonal = pow(camera_width * camera_width + camera_height * camera_height, 0.5);
+
+  int field_of_view = 78;
+  float depth = (diagonal / 2) / tan(keisan::make_degree(field_of_view).radian() / 2);
+
+  float view_h_angle = keisan::make_degree(2 * atan2(camera_width / 2, depth)).degree();
+  float view_v_angle = keisan::make_degree(2 * atan2(camera_height / 2, depth)).degree();
+
+  // There is no object with the label we want
+  if (filtered_result.empty()) {
+    object_count = 0;
+    if (no_object_count < no_object_max_count) {
+      tracking();
+      no_object_count++;
+    } else {
+      initialize();
+    }
+  } else {
+    no_object_count = 0;
+    if (object_count < object_max_count) {
+      object_count++;
+    } else {
+      keisan::Point2 center = keisan::Point2(camera_width / 2, camera_height / 2);
```

Comment:
i guess it's possible to do this `keisan::Point2(camera_width, camera_height) / 2`

Reply:
yes I think that is possible